services/fhir.ts

```diff
@@ -180,24 +266,46 @@ export async function deleteFHIRResource<R extends AidboxResource>(
         return failure({});
     }
 
-    return service({
+    return service(markAsDeleted(resource, searchParams));
+}
+
+export function markAsDeleted<R extends AidboxResource>(
+    resource: AidboxReference<R>,
+    searchParams?: SearchParams
+): AxiosRequestConfig {
+    const inactiveMappingItem = inactiveMapping[resource.resourceType];
+
+    if (!inactiveMappingItem) {
+        console.error(`Please specify inactiveMapping for ${resource.resourceType} to mark item deleted`);
+        return {};
+    }
+
+    return {
         method: 'PATCH',
         url: `/${resource.resourceType}/${resource.id}`,
         data: {
             [inactiveMappingItem.statusField]: inactiveMappingItem.value,
         },
-    });
+        ...(searchParams ? { params: searchParams } : {}),
+    };
 }
 
 export async function forceDeleteFHIRResource<R extends AidboxResource>(
     resource: AidboxReference<R>,
-    params?: SearchParams
+    searchParams?: SearchParams
 ): Promise<RemoteDataResult<R>> {
-    return service({
+    return service(forceDelete(resource, searchParams));
+}
+
+export function forceDelete<R extends AidboxResource>(
+    resource: AidboxReference<R>,
+    searchParams?: SearchParams
+): AxiosRequestConfig {
+    return {
         method: 'DELETE',
         url: `/${resource.resourceType}/${resource.id}`,
```

Comment:
Let's change url: if we have searchParams - it should be /resourceType, otherwise /resourceType/id. And throw an exception if searchParams and id both specified.